File: ios/Plugin/TestCapPlugin.m
// MyPlugin.m
#import "TestCapPlugin.h"

@implementation TestCapPlugin

- (void)echo:(CAPPluginCall *)call {
    // Access parameters using the options property
    NSDictionary *options = [call options];
    NSString *value = options[@"value"];
    
    // Your logic here
    NSString *resultString = [NSString stringWithFormat:@"Echo: %@", value];
    
    // Send the result back to JavaScript
    [call resolve:@{@"result": resultString}];
}
@end

File: ios/Plugin/TestCapPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(TestCapPlugin)
public class TestCapPlugin: CAPPlugin {
    private let implementation = TestCap()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
    
    @objc func concatenateDate(_ call: CAPPluginCall) {
        let value = call.getString("text") ?? ""
        call.resolve([
            "text": implementation.concatenateDate(value),
            "reversedString": implementation.reverseString(value)
        ])
    }
    @objc func reverseString(_ call: CAPPluginCall) {
        let value = call.getString("text") ?? ""
        call.resolve([
            "text": implementation.reverseString(value)
        ])
    }
    
}
